### PR TITLE
[Guide] Continuous Integration With Netlify Build Plugins

### PR DESCRIPTION
**Pull request by HenrikFricke**

### CODE REVIEW

File: content/blog/guides/2020/2020-04-28-continous-integration-with-netlify-build-plugins/index.mdx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Okay, so let me get this straight: This blog is deployed on Netlify because we love Netlify ❤️. Seriously, they put just a nice focus on everything you need to be a happy developer: It’s well integrated with GitHub, you get fancy preview URLs for every single pull request, you can run your website with zero configuration and beyond that they offer a great Netlify CLI and API in case you want to hack their system. So everything is great? Almost! The thing is, I don’t work on this blog alone and so we needed a continuous integration (CI) pipeline to keep our blog running smoothly. We do boring stuff like running unit tests, linting the code or check the `yarn.lock` file for uncommitted changes. As I said, boring stuff but really important. In order to get this pipeline we configured a GitHub workflow to run all the checks for every pull request. We, of course, also run the workflow with every merge on master so we ended up disabling Netlify’s great **auto publishing** feature and used a GitHub action to deploy the blog manually within the workflow. Sure it works, but it feels messy and I was super unhappy to loose the full power of the Netlify ecosystem.
```
Comment (andrew-secret) [inline content/blog/guides/2020/2020-04-28-continous-integration-with-netlify-build-plugins/index.mdx:11]: + Nice introduction but I would keep it a bit shorter.. I would be more precise about what the post is about here.
+ I would also explicitly write that we removed GithubActions it's not really clear to me in the text
+ `loose` === `lose`

File: content/blog/guides/2020/2020-04-28-continous-integration-with-netlify-build-plugins/index.mdx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	} catch (error) {
+    utils.build.failBuild('CI failed!');
+  }
+};
+```
+
+What do I do here?
+
+1. First, I check my `yarn.lock` file for uncommitted changes after installing the dependencies. That way I ensure the integrity of the dependencies.
+2. After that, we build our Storybook and then run the unit tests
```
Comment (andrew-secret) [inline content/blog/guides/2020/2020-04-28-continous-integration-with-netlify-build-plugins/index.mdx:53]: Isn't it the wrong order? And what about the linting step?
`> lint > test:ci > build-storybook`

File: content/blog/guides/2020/2020-04-28-continous-integration-with-netlify-build-plugins/index.mdx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+- Do any kind of post-build optimisation (like compression, minifying etc.)
+- …
+
+## So now everything is great?
+
+What do you think? It’s still beta and I have stuff on my wish list. I would love to see running plugins in parallel so we could speed up our build pipeline. Another thing would be to have GitHub pull request checks on a plugins bases so I can easily identify the failing plugins within my GitHub workflow and probably get the error directly in GitHub so I don’t need to scroll through a horrible debug log. But overall, I guess it’s a great step in the right direction to stay in the Netlify ecosystem and don’t mess up different tools to get things done like a simple CI pipeline or notifying you on Slack.
+
+In case you are totally excited now, here is a reading list:
+
+- [Build Plugins | Netlify Docs](https://docs.netlify.com/configure-builds/build-plugins/#enable-build-plugins-beta)
```
Comment (andrew-secret) [inline content/blog/guides/2020/2020-04-28-continous-integration-with-netlify-build-plugins/index.mdx:80]: Maybe I would shorten the links a bit here

File: content/blog/guides/2020/2020-04-28-continous-integration-with-netlify-build-plugins/index.mdx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+---
+title: Continuous Integration with Netlify Build Plugins
+date: '2020-04-28T19:15:38.655Z'
+slug: '/continuous-integration-with-netlify-build-plugins'
+desription: 'With the new Netlify build plugins it just got super simple to create a CI pipeline'
+draft: true
+---
+
+## What are you talking about?
```
Comment (andrew-secret) [inline content/blog/guides/2020/2020-04-28-continous-integration-with-netlify-build-plugins/index.mdx:9]: Should we use title case?

Comment (HenrikFricke) [reply content/blog/guides/2020/2020-04-28-continous-integration-with-netlify-build-plugins/index.mdx:9]: Done :)

**Pull request merged**